Fix biscotti on Azure

rhass opened the pull request:
This fixes a bug on Azure where the frontend fails to load. There is a set of assumptions where we set a token when authorizing an instance, however since we do not do authorization on Azure it would fail to load
the frontend.

/cc @prajaktapurohit @nsdavidson

ryancragun commented on files/biscotti/app/extensions/authorize_dsl.rb:
```diff
@@ -23,7 +23,7 @@ def handle_authorize!
         def authorize!
           halt(400) unless payload.key?("token")
-          halt(401) unless valid_token?(payload["token"])
+          halt(401) if payload["token"] != "" && !valid_token?(payload["token"])
```
I wonder if we should make `valid_token?()` return false if the string is empty rather than introducing another case that contradicts what the token checking method name implies.

rhass replied:
@ryancragun The core problem is that it does return false as `token = ""` and `hashed_token` has a value.

rhass replied:
Thinking about your suggestion more, we can just do the inverse and return `true` if the string is empty.

ryancragun replied:
I was thinking `halt(401) unless valid_token?(payload["token"])` where `valid_token?()` when given an empty string returns false.

rhass commented:
@ryancragun I rewrote this PR to enable the Authorization on Azure VMs, and added a new CLI option to make it less of a pain for users to find their unique Azure VM id since they do not show that information in the Azure portal.

rhass commented:
Example output from new CLI option:
```
azure@automatetestvm:~$ sudo chef-marketplace-ctl show-instance-id
Looking up current Instance ID...
f0a428d7-d26a-4865-ad5e-39bee2650c5b
```

rhass commented on files/chef-marketplace-ctl-commands/show_instance_id.rb:
```diff
@@ -0,0 +1,20 @@
+add_command_under_category "show-instance-id", "Configuration", "Show the cloud instance ID. Required for first time setup authorization.", 2 do
+  puts "Looking up current Instance ID..."
+
+  system = Ohai::System.new.all_plugins
```
There is probably a way to run just the Azure plugin and shave of a couple seconds, but I couldn't figure out an easy way to do this. :man_shrugging:

ryancragun replied:
`system` is a "reserved" keyword because it's in the [`kernel` module](https://ruby-doc.org/core-2.5.1/Kernel.html#method-i-system). Overloading it is generally not a good idea.

ryancragun commented on files/chef-marketplace-cookbooks/chef-marketplace/recipes/_config_biscotti.rb:
```diff
@@ -33,11 +33,20 @@
     "To begin configuring Chef Automate, enter the #{uuid_type} for the Ec2 instance. The #{uuid_type} can be found in the AWS Console."
   node.default['chef-marketplace']['biscotti']['auth_required'] = true
+when 'azure'
+  node.default['chef-marketplace']['biscotti']['message'] = "To begin configuring Chef Automate, enter the Azure unique vmId. The vmId can be found by logging into the instance and running the command: `sudo chef-marketplace-ctl show-instance-id`"
```
It's a real bummer that there's no way to see this information in the Azure portal...
